explain simple enhancer example (#4594)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/Routing/AdvancedRoutingConfiguration.rst

      CategoryListing:
        type: Simple
        limitToPages: [13]

>         routePath: '/show-by-category/{category}/{tag}'

Reviewer: ```suggestion
        routePath: '/show-by-category/{category_id}/{tag}'
```

        defaults:
          tag: ''
        requirements:

>           category: '[0-9]{1,3}'

Reviewer: ```suggestion
          category_id: '[0-9]{1,3}'
```

          category_id: '[0-9]{1,3}'
          category: '[0-9]{1,3}'
          tag: '[a-zA-Z0-9]+'
        _arguments:
          category_id: 'category'

> :yaml:`routePath` defines the static keyword and the placeholders.

Reviewer: ":aspect:" is not a directive we use anywhere.

`category_id` makes it more clear that a ID is used, instead of a "category name". We should keep that.

Explaining "tag" and "category_id" here is good, but I would rather use the sub keys to reference it.

Author: Yes, `category_id` makes it clear, but it is **wrong**! This won't work.

Reviewer 2: I have just committed 3 new versions which should solve the miss-understandings. So, IMO this is solved now.

Reviewer 2: Sorry, may false. unresolved that issue. For me `:aspect:` ist just an optical "highlight" like described here: https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/WritingReST/Reference/Code/InlineCode.html#text-roles
So yes, I also would prefer `:yaml:` here.

Reviewer 2: OK...got it, because one single word is not valid YAML we should use another directive to highlight such words. Further, syntax checks and parsing of yaml directives may fail (in future), if they contain invalid yaml.

Reviewer: I don't really understand, could you explain this to me? `:yaml:` is used many times (154x) in this manual alone, even for single words, what's wrong about it?

Reviewer 3: When you use the text role it doesnt have to be a complete valid yaml file. That is only true for code-blocks

Reviewer 3: So the yaml text role is correct here

> :aspect:`routePath` defines the static keyword and the placeholders.

Reviewer: ```suggestion
:yaml:`routePath` defines the static keyword and the placeholders.
```

> :aspect:`category` is a unique integer of the category which can be the uid in its database table.

Reviewer: ```suggestion
:yaml:`requirements.category_id` is a unique integer of the category which can be the uid in its database table.
```

> :aspect:`tag` is a string parameter with valid characters from `a .. z`, `A .. Z` and `0 .. 9`.

Reviewer: ```suggestion
:yaml:`requirements.tag` is a string parameter with valid characters from `a .. z`, `A .. Z` and `0 .. 9`.
```

          category_id: 'category'

:yaml:`routePath` defines the static keyword and the placeholders.

> :yaml:`requirements.category_id` is a unique integer of the category which can be the uid in its database table.

Reviewer 2: The example here only shows that there is an argument in URL called "category" `&category=003`. With `_arguments` a mapping to a "better" name of that argument name happens and the new name `category_id` will be checked against a reg exp. Nothing else happens here. No table lookup, no UID, no record. The number can also be the second or third category of whatever.
As long as no aspect with more details on how to "validate" that category value, it is just a 1 to 1 mapping from URL argument to a slug.

          category_id: 'category'

:yaml:`routePath` defines the static keyword and the placeholders.

> :yaml:`requirements.category_id` is a unique integer of the category.

Reviewer 2: Hi Franz. What do you think about following?

```suggestion
:yaml:`requirements.category_id` This slug is checked using a regular expression to determine whether the string can be interpreted as a number with up to 3 digits. Although the slug appears as an integer, it is treated as a character string, whereby leading zeros are possible.
```

          category_id: 'category'

:yaml:`routePath` defines the static keyword and the placeholders.

> :yaml:`requirements.category_id` is a unique integer of the category. 

Reviewer 2: ```suggestion
:yaml:`requirements.category_id` This slug is checked using a regular expression to determine whether the argument can be interpreted as a number with up to 3 digits. Although the slug appears as an integer, it is treated as a character string, whereby leading zeros are possible.
```

> Without using `_arguments` the parameter the placeholder has to exactly match the argument name from URI which is `category`.

Reviewer 2: ```suggestion
Without using `_arguments` the placeholder name has to exactly match the argument name from URI which is `category`.
```

:yaml:`requirements.tag` is a string parameter with valid characters from `a .. z`, `A .. Z` and `0 .. 9`.
:yaml:`_arguments.category_id` defines a mapping from the `category` parameter in the URL. 

> Therefore 'category' can be used in the YAML configuration instead of `category_id`. 

Reviewer 2: ```suggestion
Therefore 'category_id' can be used in the YAML configuration instead of `category`. 
```

          category_id: 'category'

:yaml:`routePath` defines the static keyword and the placeholders.
:yaml:`requirements.category_id` is a unique integer of the category. 

> Without `_arguments` the parameter must have been 'category_id' as well .

Reviewer 2: The example writes `index.php?id=13&category=241&tag=Benni`. So, without using `_arguments` the variable has to exactly match the argument name from URI. Here would be the new example code:

```
routeEnhancers:
  CategoryListing:
    type: Simple
    limitToPages: [1]
    routePath: '/show-by-category/{category}'
    requirements:
      category: '[0-9]{1,3}'
```

:yaml:`routePath` defines the static keyword and the placeholders.
:yaml:`requirements.category_id` is a unique integer of the category. 

> Without using `_arguments` the parameter the placeholder has to exactly match the argument name from URI which is `category`.

Reviewer 2: Sorry to interrupt you again. There is something duplicated in your sentence. Do you prefer "the parameter" or "the placeholder"?
